frontend/ui/data/onetime_migration.lua

```diff
@@ -585,5 +587,27 @@ if last_migration_date < 20230710 then
     end
 end
 
+-- 20230727, Statistics plugin null id_book in page_stat_data
+if last_migration_date < 20230727 then
+    logger.info("Performing one-time migration for 20230727")
+    local db_location = DataStorage:getSettingsDir() .. "/statistics.sqlite3"
+    if util.fileExists(db_location) then
+        local conn = SQ3.open(db_location)
```

Comment:
Can it happen the file exists, but SQ3.open fails ? ie. db too big, file system full, db corrupted, or fs fuckup and chkdsk makde it empty, would SQ3.open throw an exception or just go on with it ?

Reply:
We actually guard the actual production code in the plugin far less defensively than this, so I don't think we have anything to worry about ;).

Reply:
I don't want to look how precisely we pcall() plugins when loading them, and if the init() is included - in which case a bad db would just disable the statistics plugin - while this happening here would prevent KOReader from launching ?

Ok, had a look:
```diff
function ReaderStatistics:init()
+    abc = def.ghi

```
`07/28/23-18:59:59 ERROR Failed to initialize statistics plugin: plugins/statistics.koplugin/main.lua:105: attempt to index global 'def' (a nil value)`
No statistics menu.

Corrupting my statistics.sqlite3 (`dd bs=123 count=1`), the plugin loads, in spite of:
`07/28/23-19:01:51 ERROR Failed to initialize statistics plugin: common/lua-ljsqlite3/init.lua:60: ljsqlite3[corrupt] database disk image is malformed`
its menu is shown, some functions does not work, some crash, but reading and changing books work.

Putting this at top of onetime_migration.lua:
```lua
local db_location = DataStorage:getSettingsDir() .. "/statistics.sqlite3"
if util.fileExists(db_location) then
    local conn = SQ3.open(db_location)
    logger.warn("DB CORRUPTED BUT I'M OK")
end
```
have us not crash:
`07/28/23-19:07:45 WARN  DB CORRUPTED BUT I'M OK`

So, I guess it's ok if only what happens after is pcall()ed. (but who knows what happens with some other kind of corruption?)